blockchain/tx_pool.go

```diff
@@ -707,7 +707,12 @@ func (pool *TxPool) validateTx(tx *types.Transaction) error {
 		return ErrTxTypeNotSupported
 	}
 	// Reject dynamic fee transactions until EIP-1559 activates.
-	if !pool.rules.IsEthTxType && (tx.Type() == types.TxTypeEthereumDynamicFee || tx.Type() == types.TxTypeEthereumSetCode) {
+	if !pool.rules.IsEthTxType && tx.Type() == types.TxTypeEthereumDynamicFee {
+		return ErrTxTypeNotSupported
+	}
+
+	// Reject set code transactions until EIP-7600(prague) activates.
+	if !pool.rules.IsEthTxType && !pool.rules.IsPrague && tx.Type() == types.TxTypeEthereumSetCode {
```

Comment:
It's safe to use `IsPrague` only since we already checked the fork order (`EthTxType <= Prague`) at the node startup.
```suggestion
	if !pool.rules.IsPrague && tx.Type() == types.TxTypeEthereumSetCode {
```

Reply:
Yes. See https://github.com/kaiachain/kaia/blob/20be35ee65ee637b6f3b3eb7939c427ca3c5740b/params/config.go#L486.